server/models/contests/participate.js

```diff
@@ -0,0 +1,79 @@
+const { pool } = require('../database')
+
+/**
+ * @param {*} param0
+ * @param {Object} param0.params
+ * @param {String} param0.username
+ * @return {Promise}
+ */
+
+function participate({ params, username }) {
+  return new Promise((resolve, reject) => {
+    const { contest_id: contestId } = params
+    pool.getConnection((error, connection) => {
+      if (error) {
+        return reject(error)
+      }
+      connection.beginTransaction((error) => {
+        if (error) {
+          return connection.rollback(() => {
```

Comment:
No need to rollback here, because no queries would have been executed till this line